drivers/sensor/lsm6dsl/lsm6dsl.c

```diff
@@ -478,7 +478,7 @@ static inline void lsm6dsl_accel_convert(struct sensor_value *val, int raw_val,
 	/* Convert to m/s^2 */
 	dval = (double)(raw_val) * (double)sensitivity * SENSOR_G_DOUBLE / 1000;
 	val->val1 = (int32_t)dval;
-	val->val2 = (((int32_t)(dval * 1000)) % 1000) * 1000;
+	val->val2= (((int32_t)(dval * 1000000)) % 1000000);
 
```

Comment:
Please check the spacing as indicated. 
Moreover I'm confused... How is it possible that was working before? The line seems three degrees of order smaller than before. Anyway, now seems correct to me. I would suggest to get rid of SENSOR_G_DOUBLE and use the sensor_g_to_ms2() API.

Reply:
> How is it possible that was working before?

If I got this right, the previous code produced values only accurate up to a thousandth of a `m/s²`, i.e. the three least significant decimal digits in `val2` were always zero. With the new code, it's accurate up to a millionth of a `m/s²` (though it's of course questionable whether the sensor can actually provide anything close to that reliably).

Reply:
@Alessio-Parmeggiani can you squash the two commits and make use of the APIs mentioned above?

Reply:
@avisconti I added a commit where I use the suggested API. By using this new method there is no need for the original conversion I proposed in the first commit.

Reply:
Can you possibly squash the two commits, or remove the first one. I would like to see only one commit in the PR

Reply:
@avisconti Done!